src/MolSim.cpp

```diff
@@ -28,77 +36,246 @@ void calculateV();
+      delta_t = atof(optarg);
+      break;
+    case 'i':
+      input_path = std::string(optarg);
+      break;
+    case 'o':
+      output_path = std::string(optarg);
+      break;
+    case 't':
+      sparse_output = false;
+      break;
+    case 'h':
+      print_help();
+      break;
+    case 'x':
+      xyz_output = true;
+      break;
+    default:
+      fprintf(stderr, "Usage: %s [-h] help\n", argsv[0]);
+      return 1;
+    }
   }
 
   FileReader fileReader;
-  fileReader.readFile(particles, argsv[1]);
+  fileReader.readFile(particles, input_path.data());
 
+  int iteration = 0;
   double current_time = start_time;
 
-  int iteration = 0;
+  std::cout << "Starting a simulation with:\n"
+            << "\tStart time: " << start_time << "\n"
+            << "\tEnd time: " << end_time << "\n"
+            << "\tDelta: " << delta_t << "\n";
 
   // for this loop, we assume: current x, current f and current v are known
   while (current_time < end_time) {
-    // calculate new x
     calculateX();
-    // calculate new f
     calculateF();
-    // calculate new v
     calculateV();
 
     iteration++;
-    if (iteration % 10 == 0) {
+    if (sparse_output && iteration % 10 == 0)
+      plotParticles(iteration);
+    else if (!sparse_output)
       plotParticles(iteration);
-    }
     std::cout << "Iteration " << iteration << " finished." << std::endl;
-
     current_time += delta_t;
   }
 
   std::cout << "output written. Terminating..." << std::endl;
+
+  std::cout << particles.size() << std::endl;
+
+  for (auto &p : particles) {
+    std::cout << "Main Particle: " << p.toString() << std::endl;
+    for (auto &p2 : particles[p]) {
+      std::cout << p2->toString() << std::endl;
+    }
+    std::cout << std::endl;
+  }
+
   return 0;
 }
 
-void calculateF() {
-  std::list<Particle>::iterator iterator;
-  iterator = particles.begin();
+void print_help() {
+  std::cout << "Usage: MolSim [options]\n";
+  std::cout << "Options:\n";
+  std::cout << "  -h                 Show this help message\n";
+  std::cout << "  -o   <file_path>   Specify output file path\n";
+  std::cout << "  -i   <file_path>   Specify input file path\n";
+  std::cout
+      << "  -e   <end_time>    Specify how long the simulation should run\n";
+  std::cout << "  -d   <time_delta>  Specify time increments\n";
+  std::cout << "  -t                 Enable testing mode (Writes a file for "
+               "each iteration)\n";
+  std::cout << "  -x                 Output .xyz files instead of .vpu\n";
+}
 
+void calculateF() {
   for (auto &p1 : particles) {
+    std::array<double, 3> force_copy = p1.getF();
     for (auto &p2 : particles) {
-      // @TODO: insert calculation of forces here!
+      double f_x, f_y, f_z = 0;
+      double distance = std::sqrt(std::pow(p1.getX()[0] - p2.getX()[0], 2) +
+                                  std::pow(p1.getX()[1] - p2.getX()[1], 2) +
+                                  std::pow(p1.getX()[2] - p2.getX()[2], 2));
```

Comment:
look at functions provided in `ArrayUtils.h`, you might be able to use them and achieve the same thing more neatly. But otherwise, it looks correct, so not urgent.